server/app/mutations/stacks/patch.rb

```diff
@@ -0,0 +1,31 @@
+require_relative 'common'
+
+module Stacks
+  class Patch < Mutations::Command
+    include Common
+
+    required do
+      model :stack_instance, class: Stack
+    end
+
+    optional do
+      model :grid, class: Grid
+      array :labels
```

Comment:
For validation, use something like this (not 100% sure about the exact regexp, should see what docker specs for labels)

```ruby
array :labels do
  string matches: /\A[a-z0-9.-]+=/
end
```

Reply:
@SpComb it seems Docker makes it ridiculously easy to work with labels, here is the snippet from their CLI:
```go
// ValidateLabel validates that the specified string is a valid label, and returns it.
// Labels are in the form on key=value.
func ValidateLabel(val string) (string, error) {
	if strings.Count(val, "=") < 1 {
		return "", fmt.Errorf("bad attribute format: %s", val)
	}
	return val, nil
}
```

Reply:
✅ RESOLVED